QUALITY_DECLARATION.md

```diff
@@ -0,0 +1,126 @@
+This document is a declaration of software quality for the `foonathan_memory` package imported by `foonathan_memory_vendor`, based on the guidelines in [REP-2004](https://www.ros.org/reps/rep-2004.html).
+
+# `foonathan_memory` Quality Declaration
+
+This quality declaration claims that the third party package `foonathan_memory` is in the **Quality Level 4** category.
+
+Below are the rationales, notes, and caveats for this claim, organized by each requirement listed in the [Package Requirements for Quality Level 3 in REP-2004](https://www.ros.org/reps/rep-2004.html).
```

Comment:
Would you mind adding a summary section with your thoughts about what would be needed for this package to be considered Quality Level 1 in terms of ROS2 usage? You can check the one I wrote for `libyaml` [here](https://github.com/ros2/libyaml_vendor/pull/14) as an example for this. 

This section is added for external dependencies as many won't be matching 100% the [REP-2004](https://github.com/ros-infrastructure/rep/pull/218) requirements and still might be considered Quality Level 1 on a case by case basis.

Reply:
Added summary section (8) at end of the QD.